Marlin/src/feature/touch/xpt2046.cpp

```diff
@@ -77,14 +81,24 @@ uint8_t XPT2046::read_buttons() {
   const uint16_t x = uint16_t(((uint32_t(getInTouch(XPT2046_X))) * tsoffsets[0]) >> 16) + tsoffsets[1],
                  y = uint16_t(((uint32_t(getInTouch(XPT2046_Y))) * tsoffsets[2]) >> 16) + tsoffsets[3];
   if (!isTouched()) return 0; // Fingers must still be on the TS for a valid read.
-
-  if (y < 175 || y > 234) return 0;
-
-  return WITHIN(x,  14,  77) ? EN_D
-       : WITHIN(x,  90, 153) ? EN_A
-       : WITHIN(x, 166, 229) ? EN_B
-       : WITHIN(x, 242, 305) ? EN_C
-       : 0;
+  
+  // Touch within the button area simulates an encoder button
+  if (y > BUTTON_AREA_TOP && y < BUTTON_AREA_BOT)
+    return WITHIN(x,  14,  77) ? EN_D
+         : WITHIN(x,  90, 153) ? EN_A
+         : WITHIN(x, 166, 229) ? EN_B
+         : WITHIN(x, 242, 305) ? EN_C
+         : 0;
+
+  // Column and row above BUTTON_AREA_TOP
+  int8_t col = (x % (LCD_FULL_PIXEL_WIDTH) ) * (LCD_WIDTH ) / (LCD_FULL_PIXEL_WIDTH),
+         row = (y % (LCD_FULL_PIXEL_HEIGHT)) * (LCD_HEIGHT) / (BUTTON_AREA_TOP);
+
+  row = (LCD_HEIGHT) - row - 1; // TODO: Can LCD or sensor be inverted?
```

Comment:
i guess the reversed Y in (my) menus is caused there... also, we have 32 extra pixels on top, i need to press in this extra space to select the last menu item...

Reply:
In theory, XPT2046_X_CALIBRATION would solve inversion issues, but I have only one display. Can you add to chat about it? https://www.facebook.com/victormateus.oliveira/

Reply:
i need to double check ive the last version of this pull request, for some reason github PR page doesnt show the same as what i merged from your branch (im on discord tpruvot)

Reply:
arf, Tanguy on discord, X is ok, only the row is reversed... in menus

Reply:
mine: rhapsodyv#9981 on discord

Reply:
so yes, this line 97 should be fully commented... there is no logic reason to inverse the row... x and y values are supposed calibrated at this point, with top left = 0 0